### DISCUSSION TITLE
Issues configuring SSL

### DISCUSSION THREAD

**Discussion by neithercurve** [requarks/wiki#6958] [Help / Questions]
OS: Ubuntu 22.04 and Docker
Wiki.js version: 2.5.3
Database engine: PostgreSQL 15.3

Hello Community,
I'm having an issue, I can't find the config.yml to configure SSL via pfx certificate. I need to configure my local certificate and then the HTTP to HTTPS redirect. Can you pleasae give me any advice?

**Comment by alexgQQ** [comment]
Hello 👋 The pfx config option is under the ssl config section, check out the [sample config here](https://github.com/requarks/wiki/blob/main/config.sample.yml#L71). You'll have to set the provider to custom, format to pfx and then the path to wherever the file is. The HTTP to HTTPS redirect is enabled in the admin settings after the initial install.  Log in as an admin and under the SSL module there is a button to enable the redirect.

**Comment by neithercurve** [comment]
Hello @alexgQQ, thank you for the support. 
There's no such enable button under the SSL module:
![immagine](https://github.com/requarks/wiki/assets/143226272/adf2e85e-9d9e-4cc5-a53a-d375c7b9f759)

Also, do I need to mount a new config file? I wasn't the one who installed the wiki and I can't find the config.yml file under the wiki home.

**Reply by alexgQQ** [reply]
It looks like you don't have SSL enabled or a HTTPS port selected in the config file yet. Those need to be updated. You'll need to either mount a new config or update the current one if you can find it. It should always be at the root of the app installation. Maybe search the disk with a `find / -type f -name "config.yml"`.

With those set you should see something like this.
![ssl_snippet](https://github.com/requarks/wiki/assets/27295902/0a18cda7-e4f5-4655-82b1-a756dff9fc67)

**Reply by neithercurve** [reply]
Thanks @alexgQQ, I've found the config.yml file inside the container:

![immagine](https://github.com/requarks/wiki/assets/143226272/72b06e0e-202c-4a54-abe3-88c010e5f558)

Do i need to change the ssl block:

![immagine](https://github.com/requarks/wiki/assets/143226272/729e138e-8b08-443f-9909-e4196c36a613)

like this:
ssl:
  enabled: true
  port: 3443

  provider: custom

  format: pfx
  pfx: /wiki/cert.com.pfx
  passphrase: passphrase
  dhparam: null

Finally how can I reload the new config file given that is inside the container?

Thank you again for the support

**Reply by alexgQQ** [reply]
Yeah those are the basic changes you'd need to make. If you change it within the container it's likely your changes will be removed if the container or docker restarts though. From the look of that config file, it is mounted/copied from somewhere else. It would be best to change it there to persist changes.

**Reply by neithercurve** [reply]
Yeah the point is how can I start the container with the new config.yml file?
I've also tried to mount the file like this but I get a conflict error:
![immagine](https://github.com/requarks/wiki/assets/143226272/dcc3b6f2-0178-4752-90a4-2263260aa345)

**Reply by alexgQQ** [reply]
That is how you can mount the config file. The error message just says you have conflicting **container** names. Either stop the current wiki container or run another with a different name `docker run -d -p 8080:3000 --name wiki-2 --restart unless-stopped -v config.yml:/wiki/config.yml ghcr.io/requarks/wiki:2`

**Reply by neithercurve** [reply]
still, doesn't works. I couldn't be able to install the custom pfx
